bindings/pyroot/CMakeLists.txt

```diff
@@ -64,7 +64,15 @@ ROOT_ADD_CXX_FLAG(_PyROOT_FLAGS -Wno-register)
 separate_arguments(_PyROOT_FLAGS)
 
 target_compile_options(PyROOT PRIVATE ${_PyROOT_FLAGS})
-target_link_libraries(PyROOT PRIVATE ${PYTHON_LIBRARIES})
+
+if(ROOT_PYTHON_NO_LINK)
+    if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin" AND "${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
```

Comment:
What happens on Linux and Windows?

Reply:
On Linux, ignoring symbols is the default so nothing needs to be added. I'm assuming from what I've read that GCC inherits this default on macOS too, but I'm not sure.

Reply:
Unless, of course, ROOT does the right thing and adds `-Wl,--no-undefined` globally...

Reply:
Can I cancel that out with something like `-Wl,--undefined`? Or should I do what `montecarlo/pythia6/CMakeLists.txt` does and strip it out?

Reply:
I'm not sure how to make it both not link, and load the right library when being used from C++. BTW, we already have this:
```
epsft-53 root $ git grep -- -Wl.*undefined
cmake/modules/SetUpLinux.cmake:  set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--no-undefined -Wl,--hash-style=\"both\"")
cmake/modules/SetUpLinux.cmake:  set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--no-undefined")
cmake/modules/SetUpLinux.cmake:    set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--no-undefined")
cmake/modules/SetUpWindows.cmake:  set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--no-undefined")
```

Reply:
Yes, I found that. That's also how I found that the pythia6 CMake filters it out.

The loading of the library from C++ is a bit tricky, because you could put the wrong one with the same name in (or not have any), and it would fail.

Would it be possible to statically link ROOT to Python when Python is also statically linked?